unity-renderer/Assets/DCLServices/WearablesCatalogService/WearablesCatalogServiceProxy.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Cysharp.Threading.Tasks;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace DCLServices.WearablesCatalogService
+{
+    /// <summary>
+    /// This service will choose LambdasWearablesCatalogService or WebInterfaceWearablesCatalogService for requesting
+    /// wearables depending on the flag 'usingUrlParamsForDebug' received in the kernel configuration.
+    /// This will be temporal while we have to live with both services.
+    /// </summary>
+    public class WearablesCatalogServiceProxy : IWearablesCatalogService
+    {
+        private const string NOT_INITIALIZED_EXCEPTION_MESSAGE = "The proxy was not yet initialized!";
+
+        public BaseDictionary<string, WearableItem> WearablesCatalog =>
+            wearablesCatalogServiceInUse.WearablesCatalog;
+
+        private IWearablesCatalogService wearablesCatalogServiceInUse;
+        private readonly IWearablesCatalogService lambdasWearablesCatalogService;
+        private readonly IWearablesCatalogService webInterfaceWearablesCatalogService;
+        private readonly BaseDictionary<string, WearableItem> wearablesCatalog;
+        private readonly KernelConfig kernelConfig;
+        private bool isInitialized;
+
+        public WearablesCatalogServiceProxy(
+            IWearablesCatalogService lambdasWearablesCatalogService,
+            IWearablesCatalogService webInterfaceWearablesCatalogService,
+            BaseDictionary<string, WearableItem> wearablesCatalog,
+            KernelConfig kernelConfig)
+        {
+            this.lambdasWearablesCatalogService = lambdasWearablesCatalogService;
+            this.webInterfaceWearablesCatalogService = webInterfaceWearablesCatalogService;
+            this.wearablesCatalog = wearablesCatalog;
+            this.kernelConfig = kernelConfig;
+        }
+
+        public void Initialize()
+        {
+            kernelConfig.EnsureConfigInitialized().Then(config => OnKernelConfigChanged(config, null));
+            kernelConfig.OnChange += OnKernelConfigChanged;
+        }
+
+        public void Dispose()
+        {
+            kernelConfig.OnChange -= OnKernelConfigChanged;
+            wearablesCatalogServiceInUse?.Dispose();
+        }
+
+        public async UniTask<IReadOnlyList<WearableItem>> RequestOwnedWearablesAsync(string userId, int pageNumber, int pageSize, CancellationToken ct)
+        {
+            await UniTask.WaitUntil(() => isInitialized, cancellationToken: ct);
```

Comment:
instead of waiting, you could throw a `NonInitializedException`.. would that add too much complexity to handle it? same for the rest

Reply:
Mikhail suggested me to wait instead of throwing exceptions here.